src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,48 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (!herbivore.hidden && !(herbivore instanceof Carnivore)) {
+      herbivore.health -= 50;
+
+      if (herbivore.health < 1) {
+        Animal.alive.splice(
+          Animal.alive.indexOf((e) => e.name === herbivore.name),
```

Comment:
Also, using `findIndex` by the animal's name may cause some issues when two animals with the same name exist

Try this test case:
```js
test('Two same herbivore only one dies', () => {
    const olderZebra = new Herbivore('Zebra');
    const yongerZebra =  new Herbivore('Zebra');
    const lion = new Carnivore("Lion");

    lion.bite(yongerZebra);
    lion.bite(yongerZebra);

    expect(Animal.alive.includes(olderZebra))
      .toBe(true);

    expect(Animal.alive.includes(lion))
      .toBe(true);
});
```